patches/0100-Add-OpenSSL-crypto-module.patch

```diff
@@ -53,6 +59,47 @@ index 0000000000..c2c06d3bff
 +func TestUnreachableExceptTests(t *testing.T) {
 +	UnreachableExceptTests()
 +}
+diff --git a/src/crypto/internal/backend/common.go b/src/crypto/internal/backend/common.go
+new file mode 100644
+index 0000000000..d33bafd1b0
+--- /dev/null
++++ b/src/crypto/internal/backend/common.go
+@@ -0,0 +1,35 @@
++// Copyright 2022 The Go Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style
++// license that can be found in the LICENSE file.
++
++//go:build !cmd_go_bootstrap && !msan && !gocrypt
++// +build !cmd_go_bootstrap,!msan,!gocrypt
++
++package backend
++
++import (
++	"runtime"
++	"syscall"
++)
++
++func init() {
++	if v, _ := envgofips(); v == "1" {
++		if runtime.GOOS != "linux" {
++			panic("FIPS mode requested (GOFIPS=1) but not supported on " + runtime.GOOS)
++		} else if !iscgo {
++			panic("FIPS mode requested (GOFIPS=1) but not supported with cgo disabled")
++		}
++	}
++}
++
++func envgofips() (string, bool) {
```

Comment:
This should probably be `envGoFIPS`? (Matching general naming convention, and `needFIPS` later.)

Reply:
Done!